phylogenetic/rules/prepare_sequences.smk

```diff
@@ -76,7 +76,7 @@ rule filter:
             --metadata {input.metadata:q} \
             --exclude {input.exclude:q} \
             --exclude-where {params.exclude_where} \
-            --output {output.sequences:q} \
+            --output-sequences {output.sequences:q} \
```

Comment:
_(non-blocking)_

Consider adding `--output-metadata` too.

The lack of `--output-metadata`¹ means downstream rules such as [export](https://github.com/nextstrain/seasonal-cov/blob/16aedac2db7356c39b98d00eafd892bd67182ba1/phylogenetic/rules/export.smk#L9-L12) are using the original unfiltered data. Unlike `augur filter` which has `--metadata-chunk-size` and selective loading of columns, `augur export` is not designed to load large metadata files². It's fine for most pathogens (including seasonal-cov given no issues so far), but this is something to keep in mind as input data grows.

¹ I believe this stems from historical usage of a single `--output` (sequences)
² this could be improved, but even then it's still more optimal to work with the relevant subset

Reply:
It's maybe worth extending the deprecation message for `augur filter --output` to say something similar to this?

Reply:
I thought the explanation was too wordy for the actual message when running with `--output`, but I added it to the deprecation listing in https://github.com/nextstrain/augur/pull/1843